File: proxy/tests/script_engine/query_set.lua
pcall(require, "../../script_libs/lua/roxy.lua")
---@type Extension
local query_set = {
	request = function(flow)
		if flow.request.url.search_params["foo"] == "bar" then
			flow.request.url.search_params:set("foo", "baz")
		end
	end,
}
Extensions = { query_set }

File: proxy/tests/script_engine/query_set.js
/// <reference path="../../script_libs/js/index.d.ts" />
/** @type {Extension} */
const query_set = {
  request(flow) {
    if (flow.request.url.searchParams.get("foo") == "bar") {
      flow.request.url.searchParams.set("foo", "baz");
    }
  }
};
globalThis.extensions = [query_set];
